.github/workflows/run-e2e.yaml

```diff
@@ -99,11 +99,32 @@ jobs:
         run: |-
           cd e2e/
 
-          kubectl scale --replicas 1 deploy worker --namespace worker-$SANDBOX_ID
-          kubectl patch cronjobs pipeline-scale-down-job -p "{\"spec\" : {\"suspend\" : true }}" --namespace worker-$SANDBOX_ID || true
+          # Check if there are worker and pipeline deployments
+          # If not, deploy and wait
 
-          kubectl scale --replicas 1 deploy pipeline --namespace pipeline-$SANDBOX_ID
-          kubectl patch cronjobs worker-scale-down-job -p "{\"spec\" : {\"suspend\" : true }}" --namespace worker-$SANDBOX_ID || true
+          wait_for_deployment="false"
+          ready_worker=$(kubectl get deployment --namespace worker-$SANDBOX_ID -o jsonpath='{.items[0].status.readyReplicas}')
+
+          if [ -z "$ready_worker" ]; then
+            kubectl scale --replicas 1 deploy worker --namespace worker-$SANDBOX_ID
+            kubectl patch cronjobs worker-scale-down-job -p "{\"spec\" : {\"suspend\" : true }}" --namespace worker-$SANDBOX_ID || true
+
+            wait_for_deployment="true"
+          fi
+
+          ready_pipeline=$(kubectl get deployment --namespace pipeline-$SANDBOX_ID -o jsonpath='{.items[0].status.readyReplicas}')
+
+          if [ -z "$ready_pipeline" ]; then
+            kubectl scale --replicas 1 deploy pipeline --namespace pipeline-$SANDBOX_ID
+            kubectl patch cronjobs pipeline-scale-down-job -p "{\"spec\" : {\"suspend\" : true }}" --namespace pipeline-$SANDBOX_ID || true
+
+            wait_for_deployment="true"
+          fi
+
+          # Wait 5 mins for the pods to spin up before running the test
+          if [ "$wait_for_deployment" = "true" ]; then
```

Comment:
Can you add more comments to split/explain this a little bit? Like explaining the ifs, etc

Reply:
added :+1: